File: BancoPython/Banco.py



class Banco:
    empleados = [] 
    cuentas = [] 

    def agregar_empleado(self, empleado):
        self.empleados.append(empleado)

    def agregar_cuenta(self, cuenta):
        self.cuentas.append(cuenta)

    def get_empleados(self):
        return self.empleados
    
    def mostrarCuenta(self):
        for cuenta in self.cuentas:
            cuenta.mostrar_informacion()

    def mostrar_empleados(self):
        if not self.empleados:
            print("\nNo hay empleados registrados.")
            return

        print("\n*** LISTA DE EMPLEADOS ***")
        for empleado in self.empleados:
            empleado.mostrar_info()

    def mostrar_empleados_por_cuenta(self, num_cuenta):
        if not self.empleados:
            print("\nNo existen empleados registrados.")
            return

        print("\n*** EMPLEADO ***")
        for empleado in self.empleados:
            for cuenta in empleado.get_cuentas():
                if cuenta.get_account_number() == num_cuenta:
                    empleado.ver_cuentas()

    def buscar_empleado(self, num_cuenta):
        for empleado in self.empleados:
            for cuenta in empleado.get_cuentas():
                if cuenta.get_account_number() == num_cuenta:
                    return empleado
        return None

File: Tarea7_U2/src/tarea7_u2/Banco.java

package tarea7_u2;

import java.util.*;

public class Banco {
  ArrayList <Employee> empleados = new ArrayList<>();

    public Banco(Employee empleado){
        empleados.add(empleado);
    }

    public Banco() {
    }
  public void agregarEmpleado(Employee empleado) {
        empleados.add(empleado);
    }

  
    public List<Employee> getEmpleados() {
        return empleados;
    }

    public void mostrarEmpleados()
    {
        if (empleados.isEmpty()) 
        {
            System.out.println("\nNo hay empleados registrados.");
            return;
        }
        System.out.println("\n*** LISTA DE EMPLEADOS ***");
        for (Employee empleados : empleados) 
        {
            empleados.verCuentas();
            
        }
    }
    public void mostrarEmpleados(long numCuenta){
        if (empleados.isEmpty()) 
        {
            System.out.println("\nNo hay empleados registrados.");
            return;
        }
        System.out.println("\n*** LISTA DE EMPLEADOS ***");
        for (Employee empleados : empleados) 
        {
            if(empleados.account.getAccountNumber()==numCuenta)
                empleados.verCuentas();
            
        }
    }
     public Employee buscarEmpleado(long numCuenta) {
        for (Employee empleado : empleados) {
            for (BankAccount cuenta : empleado.getCuentas()) {
                if (cuenta.getAccountNumber() == numCuenta) {
                    return empleado;
                }
            }
        }
        return null;
}

  
  
    
    
    
}
